core/src/test/scala/com/softwaremill/session/SessionConfigTest.scala

```diff
@@ -0,0 +1,105 @@
+package com.softwaremill.session
+
+import java.security.{KeyPairGenerator, PrivateKey}
+import java.util.Base64
+
+import com.softwaremill.session.JwsAlgorithm.HmacSHA256
+import com.typesafe.config.ConfigValueFactory.fromAnyRef
+import com.typesafe.config.{Config, ConfigFactory}
+import org.scalatest.{FlatSpec, Matchers}
+
+class SessionConfigTest extends FlatSpec with Matchers {
+
+  def referenceConfWithSecret(serverSecret: String): Config =
+    ConfigFactory
+      .load("reference")
+      .withValue("akka.http.session.server-secret", fromAnyRef(serverSecret))
+
+  def configWith(stringValue: String): Config =
+    ConfigFactory
+      .parseString(stringValue)
+      .withFallback(referenceConfWithSecret("wh4t3v3r"))
+
+  it should "load and parse default (HS256) JWS config" in {
+    val fakeServerSecret = "f4k3S3rv3rS3cr37"
+    val fakeConfig = referenceConfWithSecret(fakeServerSecret)
+    fakeConfig.getString("akka.http.session.jws.alg") should equal("HS256")
+
+    val config = SessionConfig.fromConfig(fakeConfig)
+    config.jws.alg should equal(HmacSHA256(fakeServerSecret))
+  }
+
+  it should "load and parse HS256 JWS config" in {
```

Comment:
Does this test case not duplicate the one above?

Reply:
Previous test case asserts a default value of HS256, which makes the config backward compatible (you may not bother about this whole new `jws` config branch and your tokens will be signed using the same algorithm as before).

This test case is aimed to prove that value of `HS256` is correctly interpreted (so that if anybody would change the corresponding config value & reference configuration then the suite should report a regression).